src/main/java/org/wildfly/security/http/impl/ServerMechanismFactoryImpl.java

```diff
@@ -100,7 +101,7 @@ public HttpServerAuthenticationMechanism createAuthenticationMechanism(String me
             case CLIENT_CERT_NAME:
                 return new ClientCertAuthenticationMechanism(callbackHandler);
             case DIGEST_NAME:
-                return new DigestAuthenticationMechanism(callbackHandler, nonceManager, (String) properties.get(CONFIG_REALM), (String) properties.get(CONFIG_CONTEXT_PATH), providers);
+                return new DigestAuthenticationMechanism(callbackHandler, nonceManager, (String) properties.get(CONFIG_REALM), (String) properties.get(CONFIG_CONTEXT_PATH), (String) properties.get(CONFIG_ALGORITHM), providers);
```

Comment:
I don't think we can handle the algorithm as a single property, the problem is we may want to configure an environment where some clients understand the stronger hashes whilst some don't so we may want the mechanism to be created multiple times once for each algorithm.

We need to keep the existing DIGEST_NAME unmodified but we may want to follow a similar pattern to SASL where the algorithm becomes a part of the name - that way multiple can be specified in a preferred order.

We could consider the algorithm property being a comma separated list but if we do that it would be impossible to interleave with other mechanisms if that was desirable for any reason.